src/movies.js

```diff
@@ -1,16 +1,102 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(movies) {
+    let allDirectors = movies.map(function(movie) {
+        return movie.director;
+    });
+    const uniqueDirectors = [...new Set(allDirectors)];
+    return uniqueDirectors;
+}
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(array) {
+    return array.filter(function(movie) {
+        return movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')
+    }).length;
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movAvg){
+    if(movAvg.length === 0){
+        return 0;
+    }
+    total = 0;  
+    let rating = [];
+
+   rating =  movAvg.map(function(movie){
+        if (!movie.rate){
+                return movie.rate === 0;
+                }
+        return movie.rate;
+      });
+      const sum = rating.reduce((max, val) => {
+      return max + val;
+    }); 
+      total = (sum / movAvg.length).toFixed(2);
+      return Number.parseFloat(total);
+
+    }
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate (dramAvg) {
+    // if(dramAvg.length === 0){
+    //     return 0;
+    // }
+    // total = 0
+    // let rating = [];
+
+    // rating = dramAvg.map(funtion(movie) {
+    //     if 
+    // })
+}
+
```

Comment:
Here you can filter the origin array to find the movies that have the drama genre then reuse your function from the previous iteration